### PR TITLE
Fix issue when computing the specs of a qnode measuring a Hamiltonian

### PR DESCRIPTION
**Pull request by mariaschuld**
Currently this code throws an error:

```python
import pennylane as qml

dev = qml.device("default.qubit", wires=2)
H = qml.Hamiltonian([0.1, 0.2], [qml.PauliZ(0), qml.PauliZ(0) @ qml.PauliX(1)])
@qml.qnode(dev)
def circuit():
    return qml.expval(H)

res = qml.specs(circuit)()
print(res)
```

The reason is that `specs` (or more precisely, `tape.diagonalizing_gates`) assumes that an observable has `diagonalizing_gates` defined. The new Hamiltonian observable does not, since we do not know its diagonalisation. This is a new use case for an observable, and we need to think of a longer term solution when overhauling the operation class.

For now we fix this by returning an empty list as diagonalising gates of the Hamiltonian.

### CODE REVIEW

File: tests/test_vqe.py [hunk 1]
```diff
@@ -1452,6 +1452,23 @@ def circuit(w):
         dc = jax.grad(circuit)(w)
         assert np.allclose(dc, big_hamiltonian_grad, atol=tol)
 
+    def test_specs(self):
+        # test that the specs of a VQE circuit can be computed
```
Comment (mariaschuld) [inline tests/test_vqe.py:1456]:
```suggestion
        """Test that the specs of a VQE circuit can be computed"""
```

File: pennylane/vqe/vqe.py [hunk 2]
```diff
@@ -552,6 +552,10 @@ def queue(self, context=qml.QueuingContext):
         context.append(self, owns=tuple(self.ops))
         return self
 
+    def diagonalizing_gates(self):
+        # todo: find better solution for observables whose diagonalisation we cannot compute
+        return []
```
Comment (josh146) [inline pennylane/vqe/vqe.py:557]: I feel like `raise NotImplementedError` was a nice solution, shame that the specs forces us to define it!

**Review by josh146 — APPROVED** [review]
That was fast, thanks @mariaschuld!

Happy to approve, although I have a small thought at the back of my head - why _does_ `specs` require diagonalization? Is it easy to change `qml.specs` so that there is a try-except around checking for the diagonalization?

File: pennylane/vqe/vqe.py (hunk 2 above)
Comment (mariaschuld) [reply pennylane/vqe/vqe.py:557]: Yes, I also prefer the error, because then we can catch other problems. Ok, let me do the try-except block!

**Pull request merged**